joomla/joomla-cms
# Improve checks for important menu items in custom admin menu

Improve checks for presence of important menu items in custom admin menu.

### Summary of Changes
* Improve recovery link to bring user back to same page when switching recovery mode on/off, currently it takes the user to admin home page.
* One liner message for all the 3 important menu item for simplicity.
* Stop annoying info alert on every page when user is in menu recovery mode and show “exit recovery” as a menu item inside recovery menu parent itself.

### Testing Instructions
* Create a custom admin menu and set it as your active menu in menu module settings.
* A message stating that the important menu items Module Manager, Menu Manger and Component Container are missing should be shown including a link to enable the recovery mode.
* Creating any of these items in the custom admin menu should remove that item from the list of missing items in the warning.
* Exit recovery mode should be shown in the recovery menu when you turn on recovery mode.

### Documentation Changes Required
None
